src/Repository/PetRepository.php

```diff
@@ -20,4 +20,39 @@ public function __construct(RegistryInterface $registry)
     {
         parent::__construct($registry, Pet::class);
     }
+
+    public function searchPets(Pet $pet)
+    {
+        $queryBuilder = $this->getEntityManager()->createQueryBuilder();
+        $queryBuilder->select('p')
+            ->from('App:Pet', 'p');
+        if ($pet->getSpecies() !== null) {
+            $queryBuilder->where($queryBuilder->expr()->eq(
+                'p.species',
+                "'".$pet->getSpecies()->getId()."'"
```

Comment:
https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/reference/query-builder.html#binding-parameters-to-your-query